command/agent.go

```diff
@@ -794,13 +795,16 @@ func (c *AgentCommand) Run(args []string) int {
 			select {
 			case <-c.ShutdownCh:
 				c.UI.Output("==> Vault agent shutdown triggered")
+				// Notify systemd that the server is shutting down
+				c.notifySystemd(systemd.SdNotifyStopping)
 				// Let the lease cache know this is a shutdown; no need to evict
 				// everything
 				if leaseCache != nil {
 					leaseCache.SetShuttingDown(true)
 				}
 				return nil
 			case <-ctx.Done():
+				c.notifySystemd(systemd.SdNotifyStopping)
```

Comment:
Shouldn't we also do this in the case above, when we were asked to shutdown?

Reply:
hey @ncabatoff I'm not comprehending - does `systemd.SdNotifyStopping` need to be included in any other case?

Reply:
If any **one** of the two signals are the only mode of exit (outside Windows) then I think these two later here would suffice and void any repeats - but I'm not entirely sure TBH.

Reply:
Maybe it's simpler just to add the notifySystemd call to the end of the func, or to a defer.

Reply:
Is there any chance that a end or defer function may take longer or not ever reached? - just thinking aloud in case of processor monitors or other cases an agent may be forced quit and never actually ending.

Anyway @ncabatoff any chance you can do a PR if you think a defer may be better.

Reply:
Defer will always fire when the func returns, and it shouldn't be any slower than doing it just before a return statement.